_posts/2025-10-03-demystifying-datamovers.md

```diff
@@ -0,0 +1,93 @@
+---
+author: mattslotten
+date: 2025-10-03 16:00:09 -0400
+description: "In this blog post, we dive in on Kasten's data mover and backup target sizing"
+featured: false
+image: "/images/posts/2025-10-03-demystifying-datamovers/odyssey4.jpg"
+image_caption: ""
+layout: post
+published: true
+tags: [Sizing,Object Storage]
+title: "Demystifying Kasten's Datamover Repository Sizing"
+---
+Data movers are akin to a "secret sauce" when it comes to data protection solutions. They are essential to performing backup and restore operations and can be designed in a myriad of ways to optimize performance, scale, storage utilization, and security.  In the case of k8s application data protection solutions, we often see "off-the-shelf" Kopia in both opensource (e.g. Velero) and proprietary solutions (e.g. OADP, Dell PowerProtect), while other solutions either custom-build their own, or optimize existing data mover solutions (Kasten falls in the last two camps).
+
+And with these different datamovers come tradeoffs - whether in scale, complexity, support for various workloads, performance, storage capacity, etc all while ensuring data integrity and fidelity. As a result, getting a good estimate on the actual amount of storage consumed to protect a given workload can be difficult. Then throw in things like deduplication, compression, immutability, block and fileMode disks, and we're suddenly faced with a multivariable calculus equation.
+
+![Multivariable Calculus](/images/posts/2025-10-03-demystifying-datamovers/math_meme.webp)
+
+In this blog, we'll dive into Kasten's backup target sizing and do our best to distill this single variable into a simplified arithmetic equation.
+
+{: .alert-info }
+**TLDR**; Kasten Backup Target Size (Worst Case Scenario) = Full backup + (Retention Period * Change Rate) + (Protection Period * Change Rate) + _up to_ 20 days worth of additional data
```

Comment:
2 questions : 
1. Change rate between what, between restore point right ?
2. Is it the retention period that matters or the number of restore point for the size calculation.